src/main/java/com/icthh/xm/ms/entity/service/mail/MailService.java

```diff
@@ -310,6 +310,91 @@ void sendEmail(String to,
         }
     }
 
+    /**
+     * Send mail with raw subject, from and content.
+     *
+     * @param content     the content of email
+     * @param subject     the raw subject
+     * @param email       the to email
+     * @param from        the from email
+     */
+    @Async
+    public void sendEmailWithContent(
+        String content,
+        String subject,
+        String email,
+        String from) {
+        initAndSendEmail(TenantContextUtils.getRequiredTenantKey(tenantContextHolder.getContext()),
+            content,
+            subject,
+            email,
+            MdcUtils.generateRid(),
+            from,
+            null,
+            null);
+    }
+
+    /**
+     * Send mail with raw subject, from and content.
+     *
+     * @param content     the content of email
+     * @param subject     the raw subject
+     * @param email       the to email
+     * @param from        the from email
+     * @param attachmentFilename the name of the attachment as it will appear in the mail
+     * @param dataSource the {@code javax.activation.DataSource} to take the content from, determining the InputStream
+     * and the content type
+     */
+    @Async
+    public void sendEmailWithContentAndAttachments(
+        String content,
+        String subject,
+        String email,
+        String from,
+        String attachmentFilename,
+        InputStreamSource dataSource) {
+        initAndSendEmail(TenantContextUtils.getRequiredTenantKey(tenantContextHolder.getContext()),
+            content,
+            subject,
+            email,
+            MdcUtils.generateRid(),
+            from,
+            attachmentFilename,
+            dataSource);
+    }
+
+    private void initAndSendEmail(TenantKey tenantKey,
+                                  String content,
+                                  String subject,
+                                  String email,
+                                  String rid,
+                                  String from,
+                                  String attachmentFilename,
+                                  InputStreamSource dataSource) {
+        execForCustomRid(rid, () -> {
+            if (email == null) {
+                log.warn("Can't send email on null address for tenant: {}, email content: {}",
```

Comment:
is it a really good idea to print email content to the log when the problem is related to the address? 
Maybe the subject will be enough. Because content may flood the logs or even contain confidential information

Reply:
I totally agree with you, content will be replaced with non confidential parameters